subsys/bluetooth/host/l2cap.c

```diff
@@ -1167,25 +1180,45 @@ static void le_ecred_conn_req(struct bt_l2cap *l2cap, uint8_t ident,
 	}
 
 response:
-	buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
-				      sizeof(*rsp) + (sizeof(scid) * i));
+	if (result != BT_L2CAP_LE_SUCCESS) {
+		if (i == 0) {
+			/* All connections refused */
+			buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
+				      sizeof(*rsp) + sizeof(scid));
 
-	rsp = net_buf_add(buf, sizeof(*rsp));
-	(void)memset(rsp, 0, sizeof(*rsp));
+			rsp = net_buf_add(buf, sizeof(*rsp));
+			(void)memset(rsp, 0, sizeof(*rsp));
+
+			memset(dcid, 0, sizeof(scid));
+			rsp->result = sys_cpu_to_le16(result);
+		} else {
+			buf = l2cap_create_le_sig_pdu(buf, BT_L2CAP_ECRED_CONN_RSP, ident,
+				      sizeof(*rsp) + (sizeof(scid) * i));
 
-	if (result == BT_L2CAP_LE_ERR_UNACCEPT_PARAMS ||
-	    result == BT_L2CAP_LE_ERR_PSM_NOT_SUPP ||
-	    result == BT_L2CAP_LE_ERR_AUTHENTICATION) {
-		memset(dcid, 0, sizeof(scid) * i);
+			rsp = net_buf_add(buf, sizeof(*rsp));
+			(void)memset(rsp, 0, sizeof(*rsp));
+			memset(dcid, 0, sizeof(scid) * i);
```

Comment:
This would be some connections refused then? memset of dcid to 0 seems wrong in this case. This is where we should send back the dcids allocated.